### ISSUE TITLE
Show latest git commit hash in exam programming exercise

### ISSUE THREAD

**Issue by flodt** [ls1intum/Artemis#2893]
### Is your feature request related to a problem?
When submitting programming exercises in exams, it is, by context, not entirely clear what the latest submission built on the server is based upon - especially as you (of course) don't really get feedback during the exam.
In the regular (homework) exercises, you have the submit history as context, in programming exams though, you don't.

### Describe the solution you'd like
As a reassurance to having the submission accepted by the server, how about listing the latest accepted commit hash on the exam page?
A quick `git log` to check whether the latest commit has been successfully processed would help the UX as a student, especially while looking through the exam at the end to verify your own solution.

### Additional context
The commit hash is currently only shown after the exam was submitted, but not while working on it.

**Comment by krusche** [comment]
You can already look into Bitbucket whether your submission (or commit+push) was accepted

**Comment by flodt** [comment]
That is true, and a viable option if you have the time - but seeing as the server already has access to that data, would it be possible to list it on the exam page in Artemis as well?

Opening Bitbucket, locating the correct project and repository and then checking the commit history is likely more steps than viable for a quick check in the stressful context of an exam, especially if the student's Bitbucket account has many repositories associated with it from other Artemis-using courses.

**Comment by krusche** [comment]
We could display it, but we also would like to keep the user interface clean and simple. And there is already a lot of different information displayed on the page and there are many different buttons.

One option would be to show it in the result dialog

**Comment by flodt** [comment]
> One option would be to show it in the result dialog

That sounds like a good solution - As a modal or hover-tooltip over the result/"Build failed" indicator would also be where I as a submitting student/user would look and expect it, yes.
